internal/common/baseclient/baseclient.go

```diff
@@ -26,6 +26,15 @@ type Client interface {
 	//IIFO: OPTIONAL
 	MakePrediction() (shared.PredictionInfo, error)
 	ReceivePredictions(receivedPredictions shared.PredictionInfoDict) error
+
+	//IITO: COMPULSORY
+	RequestGift() (int, error)
```

Comment:
Do we not want to be able to say how much gift we're requesting?

Reply:
Yes, this is what this function does, the int is currently not bounded, although this could easily be added if wanted

Reply:
Ah my bad, misread it, looks good!